lib/cli-engine/file-enumerator.js

```diff
@@ -296,7 +296,7 @@ class FileEnumerator {
 
         if (stat && stat.isDirectory()) {
             return this._iterateFilesWithDirectory(
-                absolutePath,
+                absolutePath + path.sep,
```

Comment:
Would you move `+ path.sep` to L383?

```js
    if (this._isIgnoredFile(directoryPath + path.sep, options)) {
```

Because both cases of a glob and a directory go through that line. We can reduce the fixed lines to the one.